pkg/tidydns/tidydns.go

```diff
@@ -84,15 +112,248 @@ const (
+		if err != nil {
+			return 0, err
+		}
+		if strings.Contains(bodyString, fmt.Sprintf("Key (username)=(%s) already exists", username)) {
+			return 0, fmt.Errorf("username already exists")
+		}
+		return 0, fmt.Errorf(errorTidyDNS, res.Status)
+	}
+
+	var user userCreate
+	err = json.NewDecoder(res.Body).Decode(&user)
+	if err != nil {
+		return 0, err
+	}
+
+	return UserID(user.Data.Id), nil
+}
+
+func (c *tidyDNSClient) GetInternalUser(ctx context.Context, userID UserID) (*UserInfo, error) {
+	req, err := http.NewRequestWithContext(
+		ctx,
+		"GET",
+		fmt.Sprintf("%s/=/user/%s", c.baseURL, strconv.Itoa(int(userID))),
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	req.SetBasicAuth(c.username, c.password)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
+	}(res.Body)
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(errorTidyDNS, res.Status)
+	}
+
+	var user userRead
+	err = json.NewDecoder(res.Body).Decode(&user)
+	if err != nil {
+		return nil, err
+	}
+
+	var groups = make([]UserInfoGroup, 0, len(user.Groups))
+	for _, group := range user.Groups {
+		groups = append(groups, UserInfoGroup{
+			GroupName:   group.GroupName,
+			Name:        group.Name,
+			Notes:       group.Notes,
+			Id:          group.Id,
+			Description: group.Description,
+		})
+	}
+
+	modifiedDate, err := time.Parse(time.DateTime, user.ModifiedDate)
+	if err != nil {
+		return nil, err
+	}
+
+	passwordChangedDate, err := time.Parse(time.DateTime, user.PasswdChangedDate)
+	if err != nil {
+		return nil, err
+	}
+
+	var ag AuthGroup
+	switch user.AuthGroup {
+	case "User":
+		ag = AuthGroupUser
+	case "SuperAdmin":
+		ag = AuthGroupSuperAdmin
+	default:
+		return nil, fmt.Errorf("unknown auth group")
+	}
+
+	return &UserInfo{
+		ModifiedBy:        user.ModifiedBy,
+		Description:       user.Description,
+		ModifiedDate:      modifiedDate,
+		Username:          user.Username,
+		AuthGroup:         ag,
+		Name:              user.Name,
+		PasswdChangedDate: passwordChangedDate,
+		Id:                UserID(user.Id),
+		Groups:            groups,
+	}, nil
+}
+
+func (c *tidyDNSClient) UpdateInternalUser(ctx context.Context, userID UserID, password *string, description *string, authGroup *AuthGroup, userAllow []UserAllowID) error {
+	data := url.Values{}
+
+	if password != nil {
+		data.Set("epassword", *password)
+		data.Set("epassword_verify", *password)
+	}
+
+	if description != nil {
+		data.Set("description", *description)
+	}
+
+	if authGroup != nil {
+		data.Set("auth_group", strconv.Itoa(int(*authGroup)))
+	}
+
+	if userAllow != nil {
+		var userAllowFormatted []string
+		if len(userAllow) > 0 {
+			userAllowFormatted = make([]string, 0, len(userAllowFormatted))
+			for _, id := range userAllow {
+				userAllowFormatted = append(userAllowFormatted, strconv.Itoa(int(id)))
+			}
+		} else {
+			userAllowFormatted = []string{""}
+		}
```

Comment:
Like commented earlier, just use `[]string{""}` as the default and avoid the else